internal/cli/server/server.go

```diff
@@ -327,22 +365,26 @@ func (s *Server) setupMetrics(config *TelemetryConfig, serviceName string) error
 	return nil
 }
 
-func (s *Server) setupGRPCServer(addr string) error {
-	s.grpcServer = grpc.NewServer(s.withLoggingUnaryInterceptor())
-	proto.RegisterBorServer(s.grpcServer, s)
-
+func (s *Server) gRPCServerByAddress(addr string) error {
 	lis, err := net.Listen("tcp", addr)
 	if err != nil {
 		return err
 	}
 
+	return s.gRPCServerByListener(lis)
+}
+
+func (s *Server) gRPCServerByListener(listener net.Listener) error {
+	s.grpcServer = grpc.NewServer(s.withLoggingUnaryInterceptor())
+	proto.RegisterBorServer(s.grpcServer, s)
+
 	go func() {
-		if err := s.grpcServer.Serve(lis); err != nil {
+		if err := s.grpcServer.Serve(listener); err != nil {
```

Comment:
Should we handle all of this outside of the go-routine synchronously?

Reply:
Because the logs becomes contradictory in case of error.

Reply:
`s.grpcServer.Serve()` will be running forever in case if everything is fine, so if we'd wait for an error then it'll block all execution.